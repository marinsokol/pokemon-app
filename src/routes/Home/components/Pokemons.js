import React, { Fragment } from 'react'
import { Link } from 'react-router-dom'
import { arrayOf, string, func, shape, number } from 'prop-types'
import { Row, Col, Card, Icon, Pagination } from 'antd'

const Pokemons = ({ count, pokemons, onToggleFavouritePokemons, onPageChange }) => (
  <Fragment>
    <Row className="row-pokemons">
      {pokemons.map(i => (
        <Col xs={6} key={i.id}>
          <Card
            cover={<img alt={i.id} src={i.imageUrl} />}
          >
            <Link to={`/pokemon-details/${i.id}`}>
              <Row>
                <Col xs={18}>
                  <b>{i.name}</b> <br />
                  {i.supertype} <br />
                  HP: {i.hp}
                </Col>
                <Col xs={6}>
                  <Icon
                    type={i.favourited ? 'heart' : 'heart-o'}
                    onClick={(e) => {
                      e.stopPropagation()
                      e.preventDefault()
                      onToggleFavouritePokemons(i.id)
                    }}
                  />
                </Col>
              </Row>
            </Link>
          </Card>
        </Col>
      ))}
    </Row>
    {pokemons.length ? (
      <Row className="home-pagination">
        <Pagination
          defaultCurrent={1}
          pageSize={20}
          total={count}
          onChange={onPageChange}
        />
      </Row>
    ) : null}
  </Fragment>
)

Pokemons.propTypes = {
  count: number.isRequired,
  pokemons: arrayOf(
    shape({
      id: string,
      imageUrl: string,
      name: string,
      supertype: string,
      hp: string
    })
  ).isRequired,
  onToggleFavouritePokemons: func,
  onPageChange: func
}

export default Pokemons
