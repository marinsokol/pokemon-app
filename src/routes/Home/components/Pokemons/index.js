import React, { Fragment } from 'react'
import { arrayOf, string, func, shape, number } from 'prop-types'
import { Row, Col, Card, Icon, Pagination } from 'antd'

const Pokemons = ({ count, pokemons, onToggleFavouritePokemons, onPageChange, onPathChange }) => (
  <Fragment>
    <Row className="row-pokemons">
      {pokemons.map(i => (
        <Col xs={6} key={i.id}>
          <Card
            cover={<img alt={i.id} src={i.imageUrl} />}
          >
            <div onClick={onPathChange(`/pokemon-details/${i.id}`)}>
              <Row>
                <Col xs={18}>
                  <b>{i.name}</b> <br />
                  {i.supertype} <br />
                  HP: {i.hp}
                </Col>
                <Col xs={6}>
                  <Icon
                    type={i.favourited ? 'heart' : 'heart-o'}
                    onClick={(e) => {
                      e.stopPropagation()
                      e.preventDefault()
                      onToggleFavouritePokemons(i.id)
                    }}
                  />
                </Col>
              </Row>
            </div>
          </Card>
        </Col>
      ))}
    </Row>
    {pokemons.length ? (
      <Row className="home-pagination">
        <Pagination
          defaultCurrent={1}
          pageSize={20}
          total={count}
          onChange={onPageChange}
        />
      </Row>
    ) : null}
  </Fragment>
)

Pokemons.propTypes = {
  count: number.isRequired,
  pokemons: arrayOf(
    shape({
      id: string,
      imageUrl: string,
      name: string,
      supertype: string,
      hp: string
    })
  ).isRequired,
  onToggleFavouritePokemons: func,
  onPageChange: func,
  onPathChange: func
}

export default Pokemons
