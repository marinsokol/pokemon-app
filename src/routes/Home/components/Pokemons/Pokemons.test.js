import React from 'react'
import sinon from 'sinon'
import { shallow, mount } from 'enzyme'
import Pokemons from './index'

describe('Pokemons', () => {
  const pokemons = [{
    id: 'ex8-98',
    imageUrl: 'https://images.pokemontcg.io/ex8/98.png',
    name: 'Deoxys ex',
    supertype: 'PokÃ©mon',
    hp: '110'
  }]
  
  it('should be defined', () => {
    expect(Pokemons).toBeDefined()
  })

  it('should render correctly', () => {
    const tree = shallow(
      <Pokemons
        count={2}
        pokemons={pokemons}
        onPathChange={() => { }}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('simulates click to toggleFavourites', () => {
    const onClick = sinon.spy()
    const tree = mount((
      <Pokemons
        count={1}
        pokemons={pokemons}
        onPathChange={() => { }}
        onToggleFavouritePokemons={onClick}
      />
    ))
    tree.find('Icon').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })

  it('simulates click to reroute', () => {
    const onClick = sinon.spy()
    const tree = mount((
      <Pokemons
        count={1}
        pokemons={pokemons}
        onPathChange={onClick}
      />
    ))
    tree.find('Card > div').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
