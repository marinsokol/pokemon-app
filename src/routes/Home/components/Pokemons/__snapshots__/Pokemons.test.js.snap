// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons should render correctly 1`] = `
<React.Fragment>
  <Row
    className="row-pokemons"
    gutter={0}
  >
    <Col
      key="ex8-98"
      xs={6}
    >
      <Card
        cover={
          <img
            alt="ex8-98"
            src="https://images.pokemontcg.io/ex8/98.png"
          />
        }
      >
        <div>
          <Row
            gutter={0}
          >
            <Col
              xs={18}
            >
              <b>
                Deoxys ex
              </b>
               
              <br />
              Pok√©mon
               
              <br />
              HP: 
              110
            </Col>
            <Col
              xs={6}
            >
              <Icon
                onClick={[Function]}
                type="heart-o"
              />
            </Col>
          </Row>
        </div>
      </Card>
    </Col>
  </Row>
  <Row
    className="home-pagination"
    gutter={0}
  >
    <Pagination
      defaultCurrent={1}
      pageSize={20}
      prefixCls="ant-pagination"
      selectPrefixCls="ant-select"
      total={2}
    />
  </Row>
</React.Fragment>
`;
